master/reference/advanced/etcd-rbac/kubernetes-advanced.md

```diff
@@ -0,0 +1,98 @@
+before attempting to use the information listed here.
+
+## Why you might be interested in this guide
+
+- You are using Kubernetes with Calico, utilizing an etcd datastore, and want
+  to restrict the access that Calico components have in the etcd datastore.
+- You are using Kubernetes and Calico who share an etcd datastore and wish
+  to ensure that Calico and Kubernetes are unable to access each other's etcd
+  data.
+- You want to restrict the access that users of calicoctl have.
+- You want to do any combination of the above.
+
+## Components that need etcd Roles
+
+The following components need certificates with a Common Name that matches an
+etcd user that has been given appropriate roles allowing access to the key
+prefixes or paths listed or linked below.
+- kube-apiserver
+  - Read and Write access to `/registry`.
+    - With Kubernetes v1.6 the etcd v3 API is used, this can be switched
+	  to etcd v2 by passing a flag to kube-apiserver.
+	- This key prefix can be changed by passing the proper option to the
+	  kube-apiserver.
+  - The etcd user needs to be given the root role to perform compaction.
+- [cni-plugin](calico-etcdv2-paths#cni-plugin)
+- [Calico policy controller](calico-etcdv2-paths#calicokube-policy-controller)
+- [calico/node](calico-etcdv2-paths#caliconode)
+- It may also be useful to create a certificate key pair for use with
+  calicoctl, even creating specific ones for
+  [read only access](calico-etcdv2-paths#calicoctl-read-only-access),
+  [policy editor access](calico-etcdv2-paths#calicoctl-policy-editor-access),
+  and [full read/write access](calico-etcdv2-paths#calicoctl-full-readwrite-access).
+
+All certificate/key pairs that are referenced below are assumed to have been
+created for the specific component with the information above.
+
+## Kubernetes API server
+
+See [here](kubernetes#kubernetes-api-server), the configuration for
+Kubernetes does not change with the more advanced Calico setup options.
+
+## Calico components
+
+Once the certificates are generated and the users and roles have been setup
+in etcd the components using them must be configured.  Here are the same
+components listed in
+[Components that need etcd Roles](#components-that-need-etcd-roles)
+and links to their detailed configuration pages:
+- [cni-plugin]({{site.baseurl}}/{{page.version}}/reference/cni-plugin/configuration)
+- [Calico policy controller]({{site.baseurl}}/{{page.version}}/reference/policy-controller/configuration)
+- [calico/node]({{site.baseurl}}/{{page.version}}/reference/node/configuration)
+- [calicoctl]({{site.baseurl}}/{{page.version}}/reference/calicoctl/setup/etcdv2)
+
+Below are examples and suggestions when using a hosted Calico install where
+the Calico components are launched through a Kuberenetes manifest file, this
+is not required and the configuration could be achieved by configuring services
+that run outside of Kubernetes.
+
+### Single cert setup
```

Comment:
I think the distinction between single and multiple certs is not the right way to describe this.